dataprocess/dataprocess.go

```diff
@@ -0,0 +1,75 @@
+// Package dataprocess manages code related to the data-accepting process
+package dataprocess
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"image"
+	"image/png"
+	"os"
+
+	"github.com/pkg/errors"
+	pc "go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/rimage"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+func convertImageToPNG(ctx context.Context, img image.Image) ([]byte, error) {
+	if lazyImg, ok := img.(*rimage.LazyEncodedImage); ok {
+		if lazyImg.MIMEType() != rdkutils.MimeTypePNG {
+			return nil, errors.Errorf("expected mime type %v, got %T", rdkutils.MimeTypePNG, img)
+		}
+		return lazyImg.RawData(), nil
+	}
+
+	if ycbcrImg, ok := img.(*image.YCbCr); ok {
+		pngImage, err := rimage.EncodeImage(ctx, ycbcrImg, rdkutils.MimeTypePNG)
+		if err != nil {
+			return nil, err
+		}
+		return pngImage, nil
+	}
```

Comment:
Need to check for not ok here

Reply:
The way this function is meant to work is to try multiple different ways to convert to a png. We only fail if all of them fail. !ok is fine, we just fall through to the next possible way to convert the image.

Reply:
I will add a comment that explains this because I realize now that it is not clear.

Reply:
I am not sure if we do. I think this might not be checking `ok` on purpose because if both cases (`lazyImg` & `ycbcrImg`) are not `ok`, then it'll default to `png.Encode(buf, img)`.

Reply:
In the original code, we fail if we got anything other than a lazy png or YCbCr, since this would be unexpected: https://github.com/viamrobotics/rdk/blob/0128690343e5718f04292048a939a7fbfca2120e/services/slam/builtin/builtin.go#L906. This was done because decoding a png from camera server and then re-encoding resulting in changes to the image that caused orbslam to be unable to use it for feature matching (https://viam.atlassian.net/browse/RSDK-1393). We wanted to explicitly error if we got something other than a lazy png (or YCbCr, for mono webcam), since otherwise we wouldn't see an explicit error from rdk, but orbslam would log that it is unable to track.

Reply:
Thank you for the clarification! I had no idea about the history of this function. I removed this option and added a comment that explains some of this.